modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -327,11 +331,55 @@ public void start(@Nullable String cfg) {
                 doStartComponent(name, startedComponents, component);
             }
 
+            CompletableFuture<Void> upToDateMetastorageRevisionFut = new CompletableFuture<>();
+
+            metaStorageMgr.listen(MetastorageEvent.REVISION_APPLIED, new EventListener<MetastorageEventParameters>() {
+                @Override
+                public boolean notify(@NotNull MetastorageEventParameters parameters, @Nullable Throwable exception) {
+                    if (exception != null) {
+                        upToDateMetastorageRevisionFut.completeExceptionally(exception);
+
+                        return true;
+                    }
+
+                    long metastorageRevision = metaStorageMgr.revision().join();
+
+                    assert metastorageRevision >= parameters.getRevision() : IgniteStringFormatter.format(
+                            "Metastorage revision must greater than the node applied revision [msRev={}, appliedRev={}",
```

Comment:
"must be greater"
"node applied revision" - maybe - "local node applied revision"?